variables.tf

```diff
@@ -0,0 +1,80 @@
+# Nomad
+variable "nomad_provider_address" {
+  type        = string
+  description = "Nomad address"
+  default     = "http://127.0.0.1:4646"
+}
+variable "nomad_datacenters" {
+  type        = list(string)
+  description = "Nomad data centers"
+  default     = ["dc1"]
+}
+variable "nomad_namespace" {
+  type        = string
+  description = "[Enterprise] Nomad namespace"
+  default     = "default"
+}
+
+# Hive
+variable "hive_service_name" {
+  type        = string
+  description = "Hive service name"
+  default     = "hive-metastore"
+}
+
+variable "hive_container_port" {
+  type        = number
+  description = "Hive port"
+  default     = 9083
+}
+
+variable "hive_container_environment_variables" {
+  type        = list(string)
+  description = "Hive environment variables"
+  default     = [""]
+}
+
+variable "hive_bucket" {
+  type = object({
+    default     = string,
+    hive        = string
+  })
+  description = "Hive requires several minio buckets"
+}
+
+# Postgres
+variable "postgres_service" {
+  type = object({
+    service_name  = string,
+    port          = number,
+    database_name = string,
+    username      = string,
+    password      = string
+  })
+  description = "Postgres service name and port"
+  default = ({
+    service_name  = ""
+    port          = 0
```

Comment:
Why not a default port?

Reply:
And in general. Defaults that are erronous should not be defaults. These should not be dafaults if you actually need to implement something to use the module.

Reply:
Yeah. I also think it should be required without default values.